### DISCUSSION TITLE
Simulate Admob in Phone

### DISCUSSION THREAD

**Discussion by RayosElDev** [invertase/react-native-firebase#5212] [General]
Hello,
After I develop the Firebase code .. it's works well with testing ids.

Now I wish test with real admob Ids but in my phone (debug mode)

i have finding where and how put the core Admob like debuging.

Are some options like:

```
List<string> deviceIds = new List<string>();
deviceIds.Add("2077ef9a63d2b398840261c8221a0c9b");
RequestConfiguration requestConfiguration = new RequestConfiguration
    .Builder()
    .SetTestDeviceIds(deviceIds)
    .build();
```
Like in Unity etc

But here, react firebase'?
what is the steps?
thanks :D

**Comment by mikehardy** [comment]
Is there something missing in the documentation?
https://rnfirebase.io/admob/usage#test-ads

https://rnfirebase.io/reference#admob
https://rnfirebase.io/reference/admob/requestoptions
https://rnfirebase.io/reference/admob/requestoptions#testDevices

Our E2E tests also serve as good examples
https://github.com/invertase/react-native-firebase/blob/master/packages/admob/e2e/requestOptions.e2e.js

Please read the docs and try things, it will be much lower latency on getting things working vs any discussion here

**Reply by RayosElDev** [reply]
I was saying about how use that. But yes I spend all morning and its done..
I'm feel so noob with react stilll >.<

```
 requestOptions={{
            requestNonPersonalizedAdsOnly: true,
            testDevices: ["YOUr-ID"],
        }}

```
Ok enough thanks
